### DISCUSSION TITLE
Support Kubernetes ephemeral containers as `binarySource` option

### DISCUSSION THREAD

**Discussion by secustor** [renovatebot/renovate#24914] [Suggest an Idea]
### What would you like Renovate to be able to do?

Kubernetes does not support Dockershim anymore, therefore the current sidecar model in the helm chart is no longer viable.    
The current preferred solution is to use `binarySource=install`, but this in case of Kubernetes leads to a lot of traffic as the binaries are download for each run of Renovate. 

A way to solve this in Kubernetes is to use [EphemeralContainers ](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) which allows to run containers in Pods whithout defining them in the manifests  and therefore these changes are not reconciled. The original use case has been to add binaries and tools for debugging to running Pods, but this seems also a good solution to add binaries dynamically to Renovate. 

Pros: 
- uses K8 native features, instead of relying on direct communication with low level components. 
- Allows to use K8 configuration like pull through proxies instead of direct downloads of vendors.

//cc @Shegox 

### If you have any ideas on how this should be implemented, please tell us here.

Add an new executor `k8s-ephemeral-containers` which will use `kubectl` or the `ephemeral containers` endpoint of an API SDK to run an ephemeral container in the same manner as `binarySource=docker`. 

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
dind should still work without dockershim
